api/apiFunctions.test.tsx

```diff
@@ -24,294 +22,370 @@ jest.mock('./apiFunctions', () => {
-    );
-  });
-});
+  xdescribe('get all leagues', () => {
+    it('should return all leagues upon successful call', async () => {
+      const mockAllLeagues = [
+        {
+          leagueId: '1',
+          leagueName: 'League One',
+          logo: '',
+          participants: ['user1', 'user2'],
+          survivors: ['user1'],
+        },
+        {
+          leagueId: '2',
+          leagueName: 'League Two',
+          logo: '',
+          participants: ['user3', 'user4'],
+          survivors: ['user4'],
+        },
+      ];
+
+      apiFunctions.getAllLeagues.mockResolvedValue(mockAllLeagues);
+
+      const result = await apiFunctions.getAllLeagues();
+
+      expect(result).toEqual([
+        {
+          leagueId: '1',
+          leagueName: 'League One',
+          logo: '',
+          participants: ['user1', 'user2'],
+          survivors: ['user1'],
+        },
+        {
+          leagueId: '2',
+          leagueName: 'League Two',
+          logo: '',
+          participants: ['user3', 'user4'],
+          survivors: ['user4'],
+        },
+      ]);
+    });
+    it('should return error upon unsuccessful call', async () => {
+      apiFunctions.getAllLeagues.mockRejectedValue(
+        new Error('Error getting all leagues'),
+      );
 
-describe('get users id', () => {
-  it('should return user document ID', async () => {
-    const userId = '123';
-    const response = '456';
-    apiFunctions.getUserDocumentId.mockResolvedValue(response);
-    const result = await apiFunctions.getUserDocumentId(userId);
-    expect(result).toEqual(response);
-  });
-  it('should return error upon unsuccessful call', async () => {
-    apiFunctions.getUserDocumentId.mockRejectedValue(
-      new Error('Error getting user document ID'),
-    );
-    await expect(apiFunctions.getUserDocumentId('123')).rejects.toThrow(
-      'Error getting user document ID',
-    );
+      await expect(apiFunctions.getAllLeagues()).rejects.toThrow(
+        'Error getting all leagues',
+      );
+    });
   });
-});
 
-describe('add user to league', () => {
-  const response = {
-    userDocumentId: 'user123',
-    selectedLeague: 'league123',
-    selectedLeagues: ['league123'],
-    participants: ['user123', 'user456'],
-    survivors: ['user123'],
-  };
-  it('should add user to league', async () => {
-    apiFunctions.addUserToLeague.mockResolvedValue(response);
-    const result = await apiFunctions.addUserToLeague({ response });
-    expect(result).toEqual(response);
-  });
-  it('should return error upon unsuccessful call', async () => {
-    apiFunctions.addUserToLeague.mockRejectedValue(
-      new Error('Error adding user to league'),
-    );
-    await expect(apiFunctions.addUserToLeague('123', '456')).rejects.toThrow(
-      'Error adding user to league',
-    );
+  describe('get users id', () => {
+    it('should return user document ID', async () => {
+      const userId = '123';
+      const response = '456';
```

Comment:
nit: Could we initialize these variables outside so we can use them in the other tests in this file as it seems like you use these values on multiple occasions?

Reply:
none of the other tests use them right now.  maybe if they are used elsewhere then that person can move them out.